source/SAMRAI/tbox/Schedule.C

```diff
@@ -446,39 +594,42 @@ Schedule::processCompletedCommunications()
    if (d_unpack_in_deterministic_order) {
 
       // Unpack in deterministic order.  Wait for receive as needed.
+      // Deterministic order is lowest to highest recv rank
 
       int irecv = 0;
-      for (TransactionSets::iterator recv_itr = d_recv_sets.begin();
-           recv_itr != d_recv_sets.end(); ++recv_itr, ++irecv) {
-
-         int sender = recv_itr->first;
-         AsyncCommPeer<char>& completed_comm = d_coms[irecv];
-         TBOX_ASSERT(sender == completed_comm.getPeerRank());
-         completed_comm.completeCurrentOperation();
-         completed_comm.yankFromCompletionQueue();
+      for (auto& comms : d_recv_coms) {
+         auto& completed_comm = comms.second;
+         int sender = comms.first;
+         TBOX_ASSERT(sender == completed_comm->getPeerRank());
+         completed_comm->completeCurrentOperation();
+         completed_comm->yankFromCompletionQueue();
 
          MessageStream incoming_stream(
-            static_cast<size_t>(completed_comm.getRecvSize()) * sizeof(char),
+            static_cast<size_t>(completed_comm->getRecvSize()) * sizeof(char),
             MessageStream::Read,
-            completed_comm.getRecvData(),
+            completed_comm->getRecvData(),
             false /* don't use deep copy */
 #ifdef HAVE_UMPIRE
             , AllocatorDatabase::getDatabase()->getStreamAllocator()
 #endif
             );
 
          d_object_timers->t_unpack_stream->start();
-         for (Iterator recv = d_recv_sets[sender].begin();
-              recv != d_recv_sets[sender].end(); ++recv) {
-            (*recv)->unpackStream(incoming_stream);
+         for (const auto& transaction : d_recv_sets_fuseable[sender]) {
+            transaction->unpackStream(incoming_stream);
+         }
+         d_recv_fuser->launch();
+#if defined(HAVE_RAJA)
+         parallel_synchronize();
```

Comment:
Is this synchronization between fused and non-fused necessary?

Reply:
If it is necesary, what about if `d_recv_sets_fuseable[sender]` is empty?

Reply:
It shouldn't be needed if it's empty and the above loop is a no-op.  I think also the launch() call is unnecessary in that case.